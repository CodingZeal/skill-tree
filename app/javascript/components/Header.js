import React, { Component } from "react";
import Logo from "./logo.png";

class Header extends Component {
  render() {
    const {
      logged_in,
      sign_out_path,
      sign_in_path,
      current_user,
      edit_user
    } = this.props;
    return (
      <div className="white-header">
        <img id="zeal-img" className="navLogo" src={Logo} />
        <div className="navLinks">
          <a
            className="navBtn"
            id="profileBtn"
            href={(logged_in && `/profile/${current_user.id}`) || sign_in_path}
          >
            My Dashboard
          </a>
          <a
            className="navBtn"
            id="signInOut"
            href={(logged_in && sign_out) || sign_in_path}
          >
            {(logged_in && "Log Out") || "Log In"}
          </a>
        </div>
      </div>
    );
  }
}

export default Header;
