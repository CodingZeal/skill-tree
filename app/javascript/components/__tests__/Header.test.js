import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import Header from "../Header";

Enzyme.configure({ adapter: new Adapter() });

describe("<Header>", () => {
  const header = shallow(<Header />);

  it("renders without crashing", () => {
    expect(header).toMatchSnapshot();
  });
  it("log in link says Log Out when user is logged in", () => {
    header.setProps({ logged_in: true, current_user: { id: 1 } });
    expect(header.find("#signInOut").text()).toBe("Log Out");
  });
  it("log in link says Log In when user is logged out", () => {
    header.setProps({ logged_in: false, current_user: { id: 1 } });
    expect(header.find("#signInOut").text()).toBe("Log In");
  });
});
