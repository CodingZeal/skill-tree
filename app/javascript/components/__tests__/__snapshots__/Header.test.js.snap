// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders without crashing 1`] = `
<div
  className="white-header"
>
  <img
    alt="zeal logo"
    className="navLogo"
    id="zeal-img"
    src="test-file-stub"
  />
  <div
    className="navLinks"
  >
    <a
      className="navBtn"
      id="profileBtn"
    >
      My Dashboard
    </a>
    <a
      className="navBtn"
      id="signInOut"
    >
      Log In
    </a>
  </div>
</div>
`;
